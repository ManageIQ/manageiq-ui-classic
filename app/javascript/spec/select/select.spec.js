import React from 'react';
import { act } from 'react-dom/test-utils';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import FormRenderer from '@data-driven-forms/react-form-renderer';
import { FormTemplate } from '@data-driven-forms/pf3-component-mapper';
import Select from '../../components/select';

const RendererWrapper = ({ onChange }) => (
  <FormRenderer
    onSubmit={ () => {} }
    FormTemplate={ FormTemplate }
    componentMapper={{
      'select': Select,
    }}
    schema={{
      fields: [{
        component: 'select',
        name: 'selectField',
        label: 'Choose',
        options: [
          {
            "label": "Dogs",
            "value": "1"
          },
          {
            "label": "Cats",
            "value": "2"
          },
        ],
        onChange,
      }],
    }}
  />
);

describe('Select component', () => {
  it('should match the snapshot', () => {
    const wrapper = mount(<RendererWrapper/>);
    expect(toJson(wrapper.find(Select))).toMatchSnapshot();
  });

  it('should call onChange when changed', async(done) => {
    const onChange = jest.fn();
    const wrapper = mount(<RendererWrapper onChange={ onChange }/>);

    await act(async() => {
      const component = wrapper.find('SelectWithOnChange Select').at(2);
      component.prop('onChange')("2");
    });

    expect(onChange).toHaveBeenCalled();
    done();
  });
});
