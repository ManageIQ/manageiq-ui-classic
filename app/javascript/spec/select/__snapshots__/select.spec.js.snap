// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select component should match the snapshot 1`] = `
<SelectWithOnChange
  component="select"
  label="Choose"
  name="selectField"
  options={
    Array [
      Object {
        "label": "Dogs",
        "value": "1",
      },
      Object {
        "label": "Cats",
        "value": "2",
      },
    ]
  }
  placeholder="<Choose>"
>
  <Select
    component="select"
    label="Choose"
    loadingMessage="Loading..."
    name="selectField"
    options={
      Array [
        Object {
          "label": "Dogs",
          "value": "1",
        },
        Object {
          "label": "Cats",
          "value": "2",
        },
      ]
    }
    placeholder="<Choose>"
  >
    <Select
      SelectComponent={[Function]}
      invalidText=""
      labelText="Choose"
      loadOptionsChangeCounter={1}
      loadingMessage="Loading..."
      name="selectField"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      options={
        Array [
          Object {
            "label": "Dogs",
            "value": "1",
          },
          Object {
            "label": "Cats",
            "value": "2",
          },
        ]
      }
      placeholder="<Choose>"
      simpleValue={false}
      value=""
    >
      <ClearedSelect
        className=""
        closeMenuOnSelect={true}
        hideSelectedOptions={false}
        invalidText=""
        isClearable={false}
        isFetching={false}
        isSearchable={false}
        labelText="Choose"
        name="selectField"
        noOptionsMessage={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onInputChange={[Function]}
        options={
          Array [
            Object {
              "label": "Dogs",
              "value": "1",
            },
            Object {
              "label": "Cats",
              "value": "2",
            },
          ]
        }
        placeholder="<Choose>"
        value=""
      >
        <Select
          className=""
          disabled={false}
          helperText=""
          id="selectField"
          inline={false}
          invalid={false}
          invalidText=""
          labelText="Choose"
          light={false}
          name="selectField"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          value=""
        >
          <div
            className="bx--form-item"
          >
            <div
              className="bx--select"
            >
              <label
                className="bx--label"
                htmlFor="selectField"
              >
                Choose
              </label>
              <div
                className="bx--select-input__wrapper"
                data-invalid={null}
              >
                <select
                  className="bx--select-input"
                  id="selectField"
                  name="selectField"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  value=""
                >
                  <SelectItem
                    disabled={false}
                    hidden={false}
                    key="1"
                    label="Dogs"
                    text="Dogs"
                    value="1"
                  >
                    <option
                      className="bx--select-option"
                      disabled={false}
                      hidden={false}
                      label="Dogs"
                      value="1"
                    >
                      Dogs
                    </option>
                  </SelectItem>
                  <SelectItem
                    disabled={false}
                    hidden={false}
                    key="2"
                    label="Cats"
                    text="Cats"
                    value="2"
                  >
                    <option
                      className="bx--select-option"
                      disabled={false}
                      hidden={false}
                      label="Cats"
                      value="2"
                    >
                      Cats
                    </option>
                  </SelectItem>
                </select>
                <ForwardRef(ChevronDown16)
                  className="bx--select__arrow"
                >
                  <Icon
                    className="bx--select__arrow"
                    fill="currentColor"
                    height={16}
                    preserveAspectRatio="xMidYMid meet"
                    viewBox="0 0 16 16"
                    width={16}
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <svg
                      aria-hidden={true}
                      className="bx--select__arrow"
                      fill="currentColor"
                      focusable="false"
                      height={16}
                      preserveAspectRatio="xMidYMid meet"
                      viewBox="0 0 16 16"
                      width={16}
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                      />
                    </svg>
                  </Icon>
                </ForwardRef(ChevronDown16)>
              </div>
            </div>
          </div>
        </Select>
      </ClearedSelect>
    </Select>
  </Select>
</SelectWithOnChange>
`;
